sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/interfaces.scala

```diff
@@ -149,10 +149,24 @@ case class AggregateExpression(
       case PartialMerge => "merge_"
       case Final | Complete => ""
     }
-    prefix + aggregateFunction.toAggString(isDistinct)
+    val aggFuncStr = prefix + aggregateFunction.toAggString(isDistinct)
+    mode match {
+      case Partial | Complete if filter.isDefined =>
+        s"$aggFuncStr filter ${filter.get.toString}"
+      case _ =>
+        aggFuncStr
+    }
   }
 
-  override def sql: String = aggregateFunction.sql(isDistinct)
+  override def sql: String = {
+    val aggFuncStr = aggregateFunction.sql(isDistinct)
+    mode match {
+      case Partial | Complete if filter.isDefined =>
```

Comment:
another idea: if the filter is not used under some modes, can we drop it when we do things like `aggExprs.map(_.copy(mode = ABC))`? Then here we can blindly print the filter if it's there.

Reply:
Yes. It might be a cleaner, but the AS-IS PR looks safer because this is read-only update.

Reply:
Yes it's safe, but the part confuses me is why only partial and complete mode respect the filter. I still need to look at `AggUtils`, where we set the mode.

Reply:
Yea, ok. I'll check the approach, too.